app/src/main/java/app/grapheneos/camera/ui/activities/MainActivity.kt

```diff
@@ -1058,16 +1058,38 @@ open class MainActivity : AppCompatActivity(),
     private lateinit var dialog: Dialog
 
     open fun bytesToHex(bytes: ByteArray): String {
-        val hexChars = CharArray(bytes.size)
-        for (i in 0 until bytes.indices.last - 1 step 3) {
-            val v: Int = bytes[i].toInt() and 0xFF
-            hexChars[i] = hexArray[v ushr 4]
-            hexChars[i + 1] = hexArray[v and 0x0F]
-            hexChars[i + 2] = ' '
+        // Represent bytes as a grid of hex digits:
+        // Add a space between every byte
+        // Double space every 4 bytes (unless end or newline)
+        // Add a newline every 8 bytes (unless end)
+
+        var outLen = bytes.size * 3 - 1 // 2 hex digits + 1 space/newline per byte (except last)
+        outLen += bytes.size / 8 // One double space per row except the last incomplete row
+        if (bytes.size % 8 > 4) {
+            outLen += 1 // One double space for the last incomplete row, if it has >4 columns
         }
+
+        // outLen will be wrong for empty inputs
+        if (bytes.isEmpty()) return ""
```

Comment:
Can you move this higher? Ideally, it should return before `outLen` is calculated, since it doesn't make much sense to calculate `outLen` if `bytes` is empty.

Reply:
Yeah I thought the code flowed better this way even though calculating `outLen` first is useless. I'll change it though.